### PR TITLE
Maintenance access ID card

### PR DESCRIPTION
**Pull request by GuillaumePrata**
## About The Pull Request
This add a spare maintenance access ID card, a box of it at the HoP locker and if the config that gives assistants maintenance access is turned ON, they spawn with it.

I wanted to add it in more places, like random maint pool, mail options for assistants if the config is disabled, blackmarket uplink, etc, etc.
I won't do it for now because I want the bulk of the discussion be on the change for assistants and not about the blackmarket uplink being imba or something else.
## Why It's Good For The Game
The goal is to move the access into a physical object that can be interacted with by more people than the HoP, basically adding another vector of possible story telling for a shift.
This has a lot of pros and cons, some depending if you are the assistant, an antagonist, security, so I will go over a few.

<details><summary>Expand for pros/cons spam</summary>
<p>

HoP will have an easy way to give maint access when asked. We have no good onboarding for HoPs on how ID access work as none of the jobs that unlock HoP deal with this feature. 
It is always sad to see people scared of playing as HoP out of fear of being too slow/fucking up. Least with this they can give the 7 first crew members that show up at the line at the first minute of the round easy access, without having them screaming at the slow HoP. Then they can learn the console/how to make more maintenance IDs with some peace.

Antagonists will have no reason to steal the core ID of Assistants, unless they want the money or to be a dick, minor thing but assistants already don't have access to coworkers willing to assist them recover from lost equipment by opening the way to the protolathe, losing their money means their round grinds to a halt.

Security will be able to take the ID from Assistants that do minor non perma/execution worth crimes and just throw them outside, this makes an arrest be more than 3 minute AFK in a cell as assistants can try to negotiate with Sec for the ID back, ask the HoP for a new one, trade with an assistant that is not using theirs, etc, etc.

Assistants can hide their identity as "Maintenance Access Crew" instead of "Unknown", minor thing but I think it adds a nice flavor for assistants doing crimes and for a smart antagonists that have a different job, letting them throw security into a chase against assistants while they hide in plain sight.

The ID becomes a minor treasure and core for the  "Assistant job identity", like insuls, gas masks and whatever else.
Quite small in the grand scale of things, but it could be soul one day.

<p>
</details>

## Changelog
:cl: Guillaume Prata
add: Maintenance Access ID cards. A box countaining these IDs spawns at the HoP closet, making distributing maintenance access to the crew smoother at roundstart and mid emergency.
balance:  If the "Assistants have maintenance access" config is ON, assistants will spawn with one of those IDs instead of having the maintenance access added directly to their core ID.
/:cl:

### CODE REVIEW

File: code/modules/jobs/job_types/assistant.dm [hunk 1]
```diff
@@ -62,6 +63,10 @@ Assistant
 	else
 		uniform = GLOB.colored_assistant.jumpskirts[index]
 
+/datum/outfit/job/assistant/proc/give_maintenance_card(mob/libing/carbon/human/target)
```
Comment (ZephyrTFA) [inline code/modules/jobs/job_types/assistant.dm:66]:
```suggestion
/datum/outfit/job/assistant/proc/give_maintenance_card(mob/living/carbon/human/target)
```

**Comment by Draggeru** [comment]
This seems really weird and completely unnecessary, as a hop main this seems MORE tedious to deal with than just using the console. Adding maint access is just the press of a button. If people are being dicks to a new player that's an admin issue not a code issue.

Most of these problems you are listing just...arnt actually problems at all. Nobody's opening the way to the protolathe? Just hack doors five head. 

The thing with security seems moot and easy to ignore. 

If you want to hide your identity just put your ID in your bag and wear a disguise. If you're a traitor just use an agent ID.

**Comment by Draggeru** [comment]
If this is about giving the hop more to do why not create a system where the hop can sit and review paperwork in exchange for rewards instead of going the Hallmark movie route and dragging out the same song and dance to the longest possible time with needless tedium.

File: code/modules/jobs/job_types/assistant.dm (hunk 1 above)
Comment (GuillaumePrata) [reply code/modules/jobs/job_types/assistant.dm:66]: I did see this and the failing linters but I was leaving home already so I can only fix it in a few hours.
Still, thank you for pointing it out.

**Comment by GuillaumePrata** [comment]
> This seems really weird and completely unnecessary, as a hop main this seems MORE tedious to deal with than just using the console.

How is opening a box and giving someone an ID more tedious than using the console itself?

> Most of these problems you are listing just...arnt actually problems at all

That is because they are not problems, I didn't call them problems, I called them pros and cons of the new system.

**Comment by flowercuco** [comment]
i think that this works better as a loot item than a feature of the game for assistants as like, messing with the access computer is Your Entire Thing, i think that making security make the HoP do it and comply is correct and good while security being able to confiscate and toss these ids cuts that interaction out. 
 
 i think that the access part is literally the least complicated and most straightforward part of the job, if you want to mess with something to make it easier for HoPs its the printing part of it. 
 
also i know that its a small thing but i dont know about allowing every assistant who has maint access also instantly having a semi-anonymous id. not having to have the id in your hand or to get a wallet or something is a huge deal i think

**Comment by jughu** [comment]
this just is loot content then auctall hop gameplay

**Comment by necromanceranne** [comment]
Actually an assistant buff, but I'm not saying a word how it is.

**Comment by GuillaumePrata** [comment]
> if you want to mess with something to make it easier for HoPs to Deal its the printing part of it and issuing new ids that are similar to old ones. 

This PR's goal is not to improve HoP gameplay, it has a small side effect that interacts with how HoPs might decide to play.
Funny that 2 people gave Hop features suggestions here tho.

**Comment by flowercuco** [comment]
its because you talked about HoP onboarding and ease/speed/whatever in the pros/cons part of the pr which implies that you give a shit about the job so i, as someone else gives a shit about the job, am responding to your comments about it.

**Comment by Draggeru** [comment]
> > This seems really weird and completely unnecessary, as a hop main this seems MORE tedious to deal with than just using the console.
> 
> How is opening a box and giving someone an ID more tedious than using the console itself?
> 
> > Most of these problems you are listing just...arnt actually problems at all
> 
> That is because they are not problems, I didn't call them problems, I called them pros and cons of the new system.

It's tedium because it takes more time for me to leave my post, get to my locker, open it, dig around for a box, return to my post, and give someone a SECOND ID then just give someone maint access. It's tedium for the player getting the second ID because you are requiring the use of TWO IDs when the current PDA system only allows the storage of ONE ID. 

Also you say there are pros and cons but all I'm seeing are cons, all of the "pros" can already be addressed with the current system. 

Does an assistant need their maintenance access removed because of crime? Just take their maintenance access off of their ID. 

Need a disguise? Get one and hide your ID.

You are not adding something that makes anything new or interesting, nor does it streamline the process of obtaining or removing maintenance access. This really just feels like tedium for the sake of tedium.

**Comment by Serxule** [comment]
is this removing maintenance access from the HOP console and IDs and forcing it into specifically one ID? if that isnt what this PR is then i genuinely have no clue why this exists

**Comment by GuillaumePrata** [comment]
> It's tedium because it takes more time for me to leave my post, get to my locker, open it, dig around for a box, return to my post, and give someone a SECOND ID then just give someone maint access. It's tedium for the player getting the second ID because you are requiring the use of TWO IDs when the current PDA system only allows the storage of ONE ID.

Keep the box nearby your post/in your bag while working your post and you fix all the HoP side of tedium.

They will have a maintenance access ID that works as a disguise, so this isn't tedium either, just a preference from player to player, pros/cons on maint access directly on your ID and pros/cons on asking for the spare ID.

> You are not adding something that makes anything new or interesting, nor does it streamline the process of obtaining or removing maintenance access.

Hard disagree.
I'm suggesting a new system that can spark interesting interactions that just won't happen right now, as the only way to interact with the maintenance access is by asking the HoP to add it to your ID or stealing from someone else.

I would rather keep assistants build into this new system, but would be fine with cutting that end and just keep the maint ID by itself. It can be used on floor satchel, cargo drone, fishing, lavaland corpses and whatever other places we can reward players with a surprise maint access.

**Comment by GuillaumePrata** [comment]
> is this removing maintenance access from the HOP console and IDs and forcing it into specifically one ID?

No.

> if that isnt what this PR is then i genuinely have no clue why this exists

Kid named "First 2 lines on the why it is good for the game section"

**Comment by Serxule** [comment]
> The goal is to move the access that assistants have into a physical object that can be interacted with by more people than the HoP, basically adding another vector of possible story telling for a shift. This has a lot of pros and cons, some depending if you are the assistant, an antagonist, security, so I will go over a few.

as if assistants arent an integral part of ss13?

> HoP will have an easy way to give maint access when asked. We have no good onboarding for HoPs on how ID access work as none of the jobs that unlock HoP deal with this feature. It is always sad to see people scared of playing as HoP out of fear of being too slow/fucking up. Least with this they can give the 7 first crew members that show up at the line at the first minute of the round easy access, without having them screaming at the slow HoP. Then they can learn the console/how to make more maintenance IDs with some peace.

hop literally gets a paper describing how the ID console works, also, you'd rather have a hop dig through their locker for an ID instead of learning how to do their job

> Antagonists will have no reason to steal the core ID of Assistants, unless they want the money or to be a dick, minor thing but assistants already don't have access to coworkers willing to assist them recover from lost equipment by opening the way to the protolathe, losing their money means their round grinds to a halt.

no one steals assistant IDs?

> Security will be able to take the ID from Assistants that do minor non perma/execution worth crimes and just throw them outside, this makes an arrest be more than 3 minute AFK in a cell as assistants can try to negotiate with Sec for the ID back, ask the HoP for a new one, trade with an assistant that is not using theirs, etc, etc.

why the fuck should we encourage security to be shitters?

> Assistants can hide their identity as "Maintenance Access Crew" instead of "Unknown", minor thing but I think it adds a nice flavor for assistants doing crimes and for a smart antagonists that have a different job, letting them throw security into a chase against assistants while they hide in plain sight.

or an antag can just use a stolen ID or a chameleon ID

> The ID becomes a minor treasure and core for the "Assistant job identity", like insuls, gas masks and whatever else. Quite small in the grand scale of things, but it could be soul one day.

how would this become soul lmfao

**Comment by GuillaumePrata** [comment]
The last time you commented on my PR I asked you one simple question Serxule.
Which you refused to answer and started an insult game against me instead, I don't see a point on trying to have a dialogue with you here on GitHub as I know it won't go anywhere.

As much as I enjoy reading your rants, this is not Reddit. Have a nice day or night!

**Comment by Serxule** [comment]
> The last time you commented on my PR I asked you one simple question Serxule. Which you refused to answer and started an insult game against me instead

you mean the PR thats only use was validhunting and cucking over antags? thats just not what happened

**Comment by blackdav123** [comment]
can someone explain what this does in fortnite terms for me

**Comment by SkeletalElite** [comment]
This makes no sense, the only access assistant ID cards have is maint. Assistants are just going to throw out their ID and use this as their only ID. It will also make disguising effortless for assistants, since it won't be abnormal to see assistants walking around as an unknown with only their maint ID. 

>assistants already don't have access to coworkers willing to assist them recover from lost equipment by opening the way to the protolathe, losing their money means their round grinds to a halt.

If you alt click a new ID, you can use your memories to open your bank account on a new ID to recover the money. Unless whoever takes your ID actively drains it of its money and then puts it in their ID, of course. However at that point they've made an active choice to steal your money which would be no different from the circumstances of someone choosing to steal money after this change, rather than just taking the access. 

>Security will be able to take the ID from Assistants that do minor non perma/execution worth crimes and just throw them outside,

They can already do this, like I said the money is irrelevant because it can be recovered by just putting your bank number from your memories on a new ID

>How is opening a box and giving someone an ID more tedious than using the console itself?

Giving someone maint access is 3 clicks in the console and HoPs are going to continue to do that rather than hand out these because juggling 2 IDs is an annoyance that takes up extra inventory. Note that this "nerf" only applies to non assistants because assistants have no other valuable access, basically just making it a buff for them. 

>Assistants can hide their identity as "Maintenance Access Crew" instead of "Unknown"

Absolutely not a minor thing. Assistants will have no reason to actually use or wear their core ID. All assistants will appear to be "Maintenance Access Crew"

**Comment by Draggeru** [comment]
> Hard disagree. I'm suggesting a new system that can spark interesting interactions that just won't happen right now, as the only way to interact with the maintenance access is by asking the HoP to add it to your ID or stealing someone else's ID.

If I can completly avoid doing what you suggest by doing what I normally already do you have 100% failed to implement a "new" system. I see 0 reason why this changes the same process of "go up to the hop and ask for maints or steal someone else's ID." In either case. This isn't sparking anything new this is just stretching the norm out for no reason 

> I would rather keep assistants build into this new system, but would be fine with cutting that end and just keep the maint ID by itself. It can be used on floor satchel, cargo drone, fishing, lavaland corpses and whatever other places we can reward players with a surprise maint access.

Once again, if I can 100% ignore what you implement by just doing what I normally do you are not implementing it well. It sounds like you're trying to refactor maints in a very poor and unorthodox way.

**Comment by Draggeru** [comment]
I'm sorry but this PR legitimately feels like it was ai generated, I get what you're trying to do but how you're trying to do it isn't going to work how you think it will

**Comment by GuillaumePrata** [comment]
> If you alt click a new ID

To do that you need a new ID, if you can get a new ID then having your ID being stolen means nothing for any job, it's how you fix the issue...
This doesn't change my point either, the pros and cons are still the same regardless of the mechanic of getting a new ID and access at the HoP line works.

> Giving someone maint access is 3 clicks in the console and HoPs are going to continue to do that rather than hand out these because juggling 2 IDs is an annoyance that takes up extra inventory. Note that this "nerf" only applies to non assistants because assistants have no other valuable access, basically just making it a buff for them.

Juggling 2 IDs but having maint access while hiding your identity can be valuable for any job, I don't know why you are saying this is a nerf.
I didn't use the word buff/nerf anywhere on my post exactly because there is no clear buff or nerf, it is all dependent on which system you prefer and your goals on the shift.
This competitive 1v1 mentality of "You're buffing assistants/you're nerfing assistants" just ruins a lot of the sandbox in the game. This is not fortnite.

**Comment by GuillaumePrata** [comment]
> If I can completly avoid doing what you suggest by doing what I normally already do you have 100% failed to implement a "new" system. 

I'm making a new system, not deleting the old one.
Of course you can just do things the old way, it is not my goal to remove that...

**Comment by Fikou** [comment]
> can someone explain what this does in fortnite terms for me

instead of instantly deleting the structures for the building blocks around you you use the fortnite pickaxe to break them

**Comment by SkeletalElite** [comment]
>To do that you need a new ID, if you can get a new ID then having your ID being stolen means nothing for any job, it's how you fix the issue...

That's because losing your ID is supposed to be an inconvenience. Why do assistants specifically need their ID split into a "money ID" and access ID.  

The claim that "assistants already don't have access to coworkers willing to assist them recover from lost equipment by opening the way to the protolathe, losing their money means their round grinds to a halt." is dubious at best. 
It costs money to print from protolathes, so regardless of who you are, you need to get a new ID to get job equipment back unless someone literally prints everything out for you. Additionally, assistants have no responsibilities and therefore nothing to "grind to a halt"

> it is all dependent on which system you prefer and your goals on the shift.

It doesn't matter what your goal is as an assistant, there is literally 0 reason for them to have their core ID with this change. Assistants will walk around without their IDs and its near useless to everyone else because anyone else wearing this ID will be suspicious. Why would they wear it other than to disguise themselves, when it's much more convenient to just have maint added to your regular ID. 

Realistically, the only use the only way I see HoPs using the box of maint IDs is to just throw them on the floor in front of the line so anyone who wants one can just take one wordlessly. This will reduce player interaction and job content for the HoP.

**Comment by GuillaumePrata** [comment]
> Why do assistants specifically need their ID split

I didn't defend that, I just brought up pros/cons and how players might interact with the new system.
The only thing I defend is the 2 first lines on the why it is good for the game section.

**Comment by SkeletalElite** [comment]
> > Why do assistants specifically need their ID split
> 
> I didn't defend that, I just brought up pros/cons and how players might interact with the new system. The only thing I defend is the 2 first lines on the why it is good for the game section.

Okay why not just add these for maint loot/ other stuff like you mentioned but keep the assistant ID unchanged. As it is right now it only serves to anonymize assistant IDs

**Comment by GuillaumePrata** [comment]
> Okay why not just add these for maint loot/ other stuff like you mentioned but keep the assistant ID unchanged.

I'm fine with that if maintainers ask me for it. But this would remove most of the interactions the system would have so I wouldn't do it from the get go.

> As it is right now it only serves to anonymize assistant IDs

There is just a lot more to a spare ID system than this.

**Comment by Fikou** [comment]
i get what youre trying to do but i think this is a bit silly, you can just use the maint id for everything other than when your identity is concealed (although maybe you do want concealed identity, in which case its just a buff) and otherwise just adds a bit of tedium if you dont have a wallet

the idea of maint cards is neat though for jobs that dont have maint access, but the hop can just supply them with that instead of giving them a special card. i think it could work as something tied to like the chief engineer or engineering in general, something antags can borrow and whatever idk

**Pull request closed without merging**